### PR TITLE
Fix smie's indent

### PR DESCRIPTION
**Pull request by yuutayamada**
This PR fix following case

```nim
when true:
  if true: break
# <- wrong indent starts here
```

and most commit are refactoring to avoid avoid smie.el's warnings like
`;` is both neither and closer. (I seem there is no critical effect, but 
not good to see.)

### CODE REVIEW

**Comment by yuutayamada** [comment]
Should I remove highlight branch's commits? I thought it will disappear after
merging #72, but please let me know I'm doing wrong.

File: tests/indents/SMIE/block.nim [hunk 1]
```diff
@@ -26,7 +26,8 @@ proc testBlock4 =
   while true:
     var x = nil
     if x.isNil: break
-  echo "should be dedented after break"
+    echo "should be dedented after break"
```
Comment (reactormonk) [inline tests/indents/SMIE/block.nim]: So it shouldn't be dedented after break?

**Comment by yuutayamada** [comment]
I think I shouldn't. Once I thought it should be dedented, but I realized 
that we can write other statement or elif/else statement after that if.
(But if the `break` is only one token in the line, dedent next line FYI)

**Comment by reactormonk** [comment]
Then change that test to contain `break` as the only token in the line.

**Comment by yuutayamada** [comment]
There are `break` tests already in block/while/for, so I merged it.

**Comment by yuutayamada** [comment]
I changed the break's behavior if the line is `else: break`:

```nim
block:
  if true: echo "else break"
  else: break
echo "`else` with `break` should be dedented"
```

**Comment by yuutayamada** [comment]
Also I added `finally: break`. Probably no more exception? (except template/macro)

**Comment by yuutayamada** [comment]
Sorry, I was moron. Even if it's else/finally, it doesn't mean
next line should be dedented. There is still possibility to
indent as same previous line. I'm not sure what is best way...

**Comment by yuutayamada** [comment]
I made customize variable for it, so users can change the behavior.

**Comment by yuutayamada** [comment]
Sorry, for those some fixes. I didn't much check about single line including ":".
(it worked old indentation code, so). I hope getting stable...

**Pull request merged**